pkg/synopsysctl/cmd_deploy.go

```diff
@@ -134,14 +134,38 @@ var deployCmd = &cobra.Command{
 		}
 		synopsysOperatorDeployer, err := deployer.NewDeployer(restconfig)
 		if err != nil {
-			log.Errorf("Error creating Horizon Deployer for Synopsys Operator: %s", err)
+			log.Errorf("Error creating Horizon Deployer for Synopsys-Operator: %s", err)
 			return nil
 		}
-		synopsysOperatorDeployer.AddReplicationController(soperatorSpec.GetOperatorReplicationController())
-		synopsysOperatorDeployer.AddService(soperatorSpec.GetOperatorService())
-		synopsysOperatorDeployer.AddConfigMap(soperatorSpec.GetOperatorConfigMap())
-		synopsysOperatorDeployer.AddServiceAccount(soperatorSpec.GetOperatorServiceAccount())
-		synopsysOperatorDeployer.AddClusterRoleBinding(soperatorSpec.GetOperatorClusterRoleBinding())
+		synopsysOperatorComponents, err := soperatorSpec.GetComponents()
+		if err != nil {
+			log.Errorf("Error creating Horizon Components for Synopsys-Operator: %s", err)
+		}
+		for _, rc := range synopsysOperatorComponents.ReplicationControllers {
```

Comment:
Synopsysctl was changed to use Senthil's CRUDComponents (instead of these loops) on a different branch that will be merged later.